src/reducers/vis-state-updaters.js

```diff
@@ -1514,3 +1526,215 @@ export function updateAnimationDomain(state) {
     }
   };
 }
+
+/**
+ * Update the status of the editor
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {string} mode to set to editor to
+ * @return {Object} nextState
+ */
+export const setEditorModeUpdater = (state, {mode}) => ({
+  ...state,
+  editor: {
+    ...state.editor,
+    mode
+  }
+});
+
+/**
+ * Update editor features
+ * @memberof visStateUpdaters
+ * @param {Object} state `visState`
+ * @param {[Object]} features to store
+ * @return {Object} nextState
+ */
+export function setFeaturesUpdater(state, {features = []}) {
```

Comment:
regarding to the comments in this function
see [this commit](https://github.com/keplergl/kepler.gl/commit/7f927840ab976305eee3200d3aedb903ab60c0df#diff-3c42abf857bb61c5f6578c72771405d2R1556)